src/main/java/org/datadog/jenkins/plugins/datadog/clients/DatadogAgentClient.java

```diff
@@ -309,8 +482,20 @@ private boolean reinitializeAgentHttpClient(boolean force) {
                             .build())
                     .build();
 
-            traceBuildLogic = new DatadogTraceBuildLogic(agentHttpClient);
-            tracePipelineLogic = new DatadogTracePipelineLogic(agentHttpClient);
+            Set<String> supportedAgentEndpoints = fetchAgentSupportedEndpoints(INFO_NUM_RETRIES);
```

Comment:
This is only executed when the plugin is initialized, right? If so, it could happen that the customer updates de DDAgent but the plugin is using the old version (APM) endpoint.

Do we want to support a force reinitialization mechanism for the user to trigger?

Also, is there a way to know if a customer is using the old or new endpoints version?

Reply:
If the user updates the Agent we will continue using APM until they update Jenkins. I don't think it's terrible. The alternative would be to check the `/info` endpoint each time. This would also solve the problem we discussed in the thread above.